syntax = "proto3";

package convertidor_de_archivos;

message ErrorGeneral {
  string error = 1;
  string mensaje = 2;
}

message InformacionArchivo {
  string idElemento = 1;
  string extension = 2;
  string hash256 = 3;
}

message SolicitudConvertirCancion {
  InformacionArchivo informacionArchivo = 1;
  bytes data = 2;
}

message SolicitudConvertirPortada {
  InformacionArchivo informacionImagen = 1;
  bytes data = 2;
}

message RespuestaImagenConvertida {
  InformacionArchivo informacionImagen = 1;
  bytes data = 2;
}

message RespuestaCancionConvertida {
  InformacionArchivo informacionArchivo = 1;
  bytes data = 2;
}

message RespuestaCancionesConvertidas {
  RespuestaCancionConvertida cancionCalidadAlta = 1;
  RespuestaCancionConvertida cancionCalidadMedia = 2;
  RespuestaCancionConvertida cancionCalidadBaja = 3;
  ErrorGeneral error = 4;
}

message RespuestaImagenesConvertidas {
  RespuestaImagenConvertida imagenCalidadAlta = 1;
  RespuestaImagenConvertida imagenCalidadMedia = 2;
  RespuestaImagenConvertida imagenCalidadBaja = 3;
  ErrorGeneral error = 4;
}

service ConvertidorDeCanciones{
  rpc ConvertirCancionAMP3(stream SolicitudConvertirCancion) returns (stream RespuestaCancionesConvertidas) {};
}

service ConvertidorDeImagenes{
  rpc ConvertirImagenAPng(stream SolicitudConvertirPortada) returns (stream RespuestaImagenesConvertidas) {};
}

